FreeBSD support

```
   Compiling gaol v0.0.1 (https://github.com/servo/gaol#227ff0ba)
error[E0432]: unresolved import `platform::process`
  --> /path/to/servo/.cargo/git/checkouts/gaol-c3a1e33941376f84/227ff0b/sandbox.rs:13:25
   |
13 | use platform::process::{self, Process};
   |                         ^^^^ no `process` in `platform`

error[E0432]: unresolved import `platform::process`
  --> /path/to/servo/.cargo/git/checkouts/gaol-c3a1e33941376f84/227ff0b/sandbox.rs:13:15
   |
13 | use platform::process::{self, Process};
   |               ^^^^^^^ Could not find `process` in `platform`

error[E0432]: unresolved import `platform::ChildSandbox`
  --> /path/to/servo/.cargo/git/checkouts/gaol-c3a1e33941376f84/227ff0b/sandbox.rs:22:20
   |
22 | pub use platform::{ChildSandbox, Sandbox};
   |                    ^^^^^^^^^^^^ no `ChildSandbox` in `platform`

error[E0432]: unresolved import `platform::Sandbox`
  --> /path/to/servo/.cargo/git/checkouts/gaol-c3a1e33941376f84/227ff0b/sandbox.rs:22:34
   |
22 | pub use platform::{ChildSandbox, Sandbox};
   |                                  ^^^^^^^ no `Sandbox` in `platform`

error[E0412]: cannot find type `Operation` in module `platform`
  --> /path/to/servo/.cargo/git/checkouts/gaol-c3a1e33941376f84/227ff0b/profile.rs:99:32
   |
99 |     PlatformSpecific(platform::Operation),
   |                                ^^^^^^^^^ not found in `platform`
   |
help: possible candidate is found in another module, you can import it into scope
   |
13 | use profile::Operation;
   |

error[E0599]: no method named `support` found for type `&profile::Operation` in the current scope
   --> /path/to/servo/.cargo/git/checkouts/gaol-c3a1e33941376f84/227ff0b/profile.rs:132:29
    |
132 |             match operation.support() {
    |                             ^^^^^^^
    |
    = help: items from traits can only be used if the trait is implemented and in scope
    = note: the following trait defines an item `support`, perhaps you need to implement it:
            candidate #1: `profile::OperationSupport`

error: aborting due to 6 previous errors

error: Could not compile `gaol`.
```

--- a/lib.rs
+++ b/lib.rs
@@ -21,14 +21,18 @@ pub mod platform {
     pub use platform::linux::{ChildSandbox, Operation, Sandbox};
     #[cfg(target_os="macos")]
     pub use platform::macos::{ChildSandbox, Operation, Sandbox};
-    #[cfg(any(target_os="android", target_os="linux", target_os="macos"))]
+    #[cfg(target_os="freebsd")]
+    pub use platform::freebsd::{ChildSandbox, Operation, Sandbox};
+    #[cfg(any(target_os="android", target_os="linux", target_os="macos", target_os="freebsd"))]
     pub use platform::unix::process::{self, Process};
 
     #[cfg(any(target_os="android", target_os="linux"))]
     pub mod linux;
     #[cfg(target_os="macos")]
     pub mod macos;
-    #[cfg(any(target_os="android", target_os="linux", target_os="macos"))]
+    #[cfg(target_os="freebsd")]
+    pub mod freebsd;
+    #[cfg(any(target_os="android", target_os="linux", target_os="macos", target_os="freebsd"))]
     pub mod unix;
 }
 

--- a/platform/freebsd/mod.rs
+++ b/platform/freebsd/mod.rs
@@ -1,0 +1,78 @@
+// Copyright 2015 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! Sandboxing on FreeBSD via Capsicum.
+
+use platform::unix::process::Process;
+use profile::{self, OperationSupport, OperationSupportLevel, Profile};
+use sandbox::{ChildSandboxMethods, Command, SandboxMethods};
+
+use libc::c_int;
+use std::io;
+
+impl OperationSupport for profile::Operation {
+    fn support(&self) -> OperationSupportLevel {
+        match *self {
+            profile::Operation::SystemInfoRead =>
+                OperationSupportLevel::AlwaysAllowed,
+            _ => OperationSupportLevel::NeverAllowed
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+pub enum Operation { }
+
+pub struct Sandbox {
+    profile: Profile,
+}
+
+impl Sandbox {
+    pub fn new(profile: Profile) -> Sandbox {
+        Sandbox {
+            profile: profile,
+        }
+    }
+}
+
+impl SandboxMethods for Sandbox {
+    fn profile(&self) -> &Profile {
+        &self.profile
+    }
+
+    fn start(&self, command: &mut Command) -> io::Result<Process> {
+        command.env("GAOL_CHILD_PROCESS", "1").spawn()
+    }
+}
+
+pub struct ChildSandbox {
+}
+
+impl ChildSandbox {
+    pub fn new(_profile: Profile) -> ChildSandbox {
+        ChildSandbox {
+        }
+    }
+}
+
+impl ChildSandboxMethods for ChildSandbox {
+    fn activate(&self) -> Result<(),()> {
+        if unsafe { cap_enter() } == 0 {
+            Ok(())
+        } else {
+            error!("Failed to init sandbox");
+            Err(())
+        }
+    }
+}
+
+extern {
+    fn cap_enter() -> c_int;
+}

